socorro/external/es/supersearch.py

```diff
@@ -450,24 +467,25 @@ def get(self, **kwargs):
                     shards = None
                     break
 
-            except RequestError as exc:
+            except BadRequestError as exc:
                 # Try to handle it gracefully if we can find out what
                 # input was bad and caused the exception.
                 # Not an ElasticsearchParseException exception
                 with suppress(IndexError):
-                    bad_input = ELASTICSEARCH_PARSE_EXCEPTION_REGEX.findall(exc.error)[
-                        -1
-                    ]
-                    # Loop over the original parameters to try to figure
-                    # out which *key* had the bad input.
-                    for key, value in kwargs.items():
-                        if value == bad_input:
-                            raise BadArgumentError(key) from exc
-
-                # If it's a search parse exception, but we don't know what key is the
-                # problem, raise a general BadArgumentError
-                if "Failed to parse source" in str(exc):
-                    raise BadArgumentError("Malformed supersearch query.") from exc
+                    if (
+                        exc.body["error"]["type"] == "x_content_parse_exception"
+                        and exc.body["error"]["caused_by"]["type"]
+                        == "illegal_argument_exception"
+                    ):
+                        bad_input = ELASTICSEARCH_PARSE_EXCEPTION_REGEX.findall(
+                            exc.body["error"]["caused_by"]["reason"]
+                        )[-1]
+
+                        # Loop over the original parameters to try to figure
+                        # out which *key* had the bad input.
+                        for key, value in kwargs.items():
+                            if value == bad_input:
+                                raise BadArgumentError(key) from exc
```

Comment:
Can you tell me more about the changes in this section and the below section? It seems like the original code handled two different kinds of errors and the new code only handles one of those here and the other one in a different block. Is that right?

Reply:
That is correct. This used to handle malformed query and invalid regex, but now it only handles malformed query, and bad regex is now considered a shard failure by ES, so it's handled below.